src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -695,6 +779,15 @@ private static DetailAST getNextToken(DetailAST ast) {
             DetailAST parent = ast;
             while (next == null && parent != null) {
                 next = parent.getNextSibling();
+                if (next != null && next.getType() == TokenTypes.EMPTY_STAT
+                        && ast.getType() == TokenTypes.LITERAL_SWITCH) {
+                    next = next.getNextSibling();
```

Comment:
I was not expecting changes to this method when I asked if we can switch to using it. Changing this affects all other options and regression will be needed to show no harm.

Why do we need these extra checks?
These look like from https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1111120766 , which are from expressions which we cannot validate.

Reply:
will remove all the support for expression in the next push

Reply:
It is possible that empty stat is at the end in switch case statement as well

removed all the expression
done

Reply:
Then it is not done if `EMPTY_STAT` is still here. 
>  was not expecting changes to this method

Why is `EMPTY_STAT` needed here when it wasn't before? How are you sure this is affecting functionality of the other checks?

Reply:
Ok I have removed the EMPTY_STAT 
I have not removed it because their are some statement which contain unnecessary `;` after rightcurly to validate that case it was not removed 

but I have removed know no other functionality is added. 
Done

Reply:
Please review further